applications/vessel-history/src/features/vessels/activity/vessels-activity.selectors.ts

```diff
@@ -0,0 +1,50 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { RenderedEvent, selectEventsWithRenderingInfo } from 'features/vessels/activity/vessels-activity.slice';
+import { Filters, initialState, selectFilters } from './../../profile/filters/filters.slice';
+
+export const selectFilteredEvents = createSelector(
+  [selectEventsWithRenderingInfo, selectFilters],
+  (events, filters) => {
+    const startTimestamp = new Date(filters.start).getTime()
+    const endTimestamp = new Date(filters.end).getTime()
+
+    return events.flat().filter((event: RenderedEvent) => {
+      if (startTimestamp > event.start || endTimestamp < event.end) {
```

Comment:
Include the event if took place in the same timestamp of start or end too
```suggestion
      if (startTimestamp >= event.start || endTimestamp <= event.end) {
```